cmd/bee/cmd/start.go

```diff
@@ -151,8 +151,14 @@ func (c *command) initStartCmd() (err error) {
 
 			// Wait for termination or interrupt signals.
 			// We want to clean up things at the end.
-			interruptChannel := make(chan os.Signal, 1)
-			signal.Notify(interruptChannel, syscall.SIGINT, syscall.SIGTERM)
+			sysInterruptChannel := make(chan os.Signal, 1)
+			signal.Notify(sysInterruptChannel, syscall.SIGINT, syscall.SIGTERM)
+
+			interruptChannel := make(chan struct{})
+			go func() {
+				<-sysInterruptChannel
+				close(interruptChannel)
```

Comment:
nice. just one last suggestion. the thing is that the structure of the interrupt interception below was structured such that:
- user presses ctrl-c to quit
- process gets stuck(??) / shutdown commences
- user presses ctrl-c again to force the shutdown and not wait for a clean shutdown OR wait for a clean shutdown

that's why there's two reads from the signal channel later down in the file. so my suggestion would be, to keep those as they are, and just move the `close` here to line 223 (and get rid of the extra goroutine). the other consequence is that right now you will never wait for the `done` to select on line 245, since the interrupt channel is already closed and will most probably always select before the `done` gets closed. details... :)